h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -2305,10 +2307,14 @@ protected void updateProgress(GLMModel fixedModel, GLMModel[] randModels, Frame
                                   double[] VC1, double[][] VC2, double sumDiff2, double convergence, boolean canScore,
                                   double[][] cholR, Frame augXZ) {
       _sc.addIterationScore(_state._iter, _state._sumEtaSquareConvergence);
-      if(canScore && (_parms._score_each_iteration || timeSinceLastScoring() > _scoringInterval)) {
+      if(canScore && (_parms._score_each_iteration || timeSinceLastScoring() > _scoringInterval || 
+              ((_parms._score_iteration_interval > 0) && ((_state._iter % _parms._score_iteration_interval) == 0)))) {
         _model.update(_state.expandBeta(_state.beta()), _state.ubeta(),-1, -1, _state._iter);
         scoreAndUpdateModelHGLM(fixedModel, randModels, glmmmeReturns, hvDataOnly, VC1, VC2, sumDiff2, convergence, 
                 cholR, augXZ, false);
+        _earlyStop = ScoreKeeper.stopEarly(_model.scoreKeepers(),
```

Comment:
I would suggest to change it to `_earlyStop = _earlyStop || ...;` to avoid unnecessary method evaluation if the `_earlyStop` is already `true`

Reply:
Nice optimization!